auth/httptransport/trace.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httptransport
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"go.opencensus.io/trace"
+	"go.opencensus.io/trace/propagation"
+)
+
+const (
+	httpHeaderMaxSize = 200
+	cloudTraceHeader  = `X-Cloud-Trace-Context`
+)
+
+var _ propagation.HTTPFormat = (*httpFormat)(nil)
+
+// httpFormat implements propagation.httpFormat to propagate
+// traces in HTTP headers for Google Cloud Platform and Cloud Trace.
+type httpFormat struct{}
+
+// SpanContextFromRequest extracts a Cloud Trace span context from incoming requests.
+func (f *httpFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanContext, ok bool) {
```

Comment:
I believe this entire file is identical to (and likely originally copied from) https://github.com/census-instrumentation/opencensus-go/blob/master/exporter/stackdriver/propagation/http.go. Just wondering if there is a better way to implement this support for context propagation. Most obviously, why do we need to support extracting Cloud Trace span context from incoming requests in client libraries? This appears to be server code.

Reply:
You may be right, I copied most of it over from https://github.com/googleapis/google-api-go-client/blob/main/transport/http/internal/propagation/http.go.

From my understanding you need to extract the header to create a base span for this realm of the application. Because it is extracted here, any future spans created from something like in storage will have this extracted context as the parent. I believe it is all about chaining the spans together.

Reply:
Discussed more offline, Chris is right we likely don't need this. But not removing it as of right now to be safe as it was in the old impl.